src/test/scala/dropInconsistentAssignments.scala

```diff
@@ -72,92 +72,122 @@ import ohnosequences.mg7._
 import com.github.tototoshi.csv._
 import better.files._
 
-case object dropInconsistentAssignments extends FilterDataFrom(dropRedundantAssignments)(deps = mg7results, ncbiTaxonomyBundle) {
+case object dropInconsistentAssignments extends FilterDataFrom(dropRedundantAssignments)(
+  deps = clusteringResults, ncbiTaxonomyBundle
+) {
 
-  type ID     = String
-  type Taxa   = String
-  type Fasta  = FASTA.Value
+  lazy val taxonomyGraph = ncbiTaxonomyBundle.graph
 
-  private lazy val taxonomyGraph = ncbiTaxonomyBundle.graph
+  /* Mapping of sequence IDs to the list of their taxonomic assignments */
+  lazy val referenceMap: Map[ID, Seq[Taxon]] = source.table.csvReader.iterator
+    .foldLeft(Map[ID, Seq[Taxon]]()) { (acc, row) =>
+      acc.updated(
+        row(0),
+        row(1).split(';').map(_.trim).toSeq
+      )
+    }
 
-  def filterData(): Unit = {
+  /* Mapping of sequence IDs to corresponding FASTA sequences */
+  lazy val id2fasta: Map[ID, Fasta] = source.fasta.stream
+    .foldLeft(Map[ID, Fasta]()) { (acc, fasta) =>
+      acc.updated(
+        fasta.getV(header).id,
+        fasta
+      )
+    }
+
+  def referenceTaxaFor(id: ID): Seq[Taxon] = referenceMap.get(id).getOrElse(Seq())
+
+  // TODO: this should be a piece of reusable code in MG7
+  def getAccumulatedCounts(taxa: Seq[Taxon]): Map[Taxon, (Int, Seq[Taxon])] = {
+    // NOTE: this mutable map is used in getLineage for memoization of the results that we get from the DB
+    val lineageMap: scala.collection.mutable.Map[Taxon, Taxa] = scala.collection.mutable.Map()
+
+    def getLineage(id: Taxon): Taxa = lineageMap.get(id).getOrElse {
+      val lineage = taxonomyGraph.getTaxon(id)
+        .map{ _.ancestors }.getOrElse( Seq() )
+        .map{ _.id }
+      lineageMap.update(id, lineage)
+      lineage
+    }
 
-    /* First we read what we've got from MG7 */
-    val id2mg7lca: Map[ID, Taxa] = mg7LCAfromFile(mg7results.lcaTable)
+    val counter = loquats.countDataProcessing()
 
-    /* Then we process the source table and compare assignments with LCA from MG7. We know that in the output of dropRedundantAssignments table and fasta IDs are synchronized, so we can just zip them. */
-    ( source.table.csvReader.iterator zip
-      source.fasta.stream.iterator
-    ).foreach {
+    val directMap      = counter.directCounts(taxa, getLineage)
+    val accumulatedMap = counter.accumulatedCounts(directMap, getLineage)
 
-      case (row, fasta) => {
+    accumulatedMap
+  }
+
+  /* This constant determines how many levels up from the considered node will be the ancestor that we want to take for comparing their cumulative counts */
+  // FIXME: this constant needs a review
+  val ancestorLevel: Int = 1 // i.e. direct parent by default
+
+  /* This threshold determines minimum percentage of the cumulative count of the considered taxon from its ancestor's count */
+  // FIXME: this constant needs a review
+  val countsPercentageMinimum: Double = 42.75
```

Comment:
Yes, I said it in https://github.com/ohnosequences/db.rna16s/pull/56#issuecomment-249992482